### PR TITLE
fix: Migrate API usage from *.metafi & *.metaswap subdomains to *.cx.metamsk.io"

### LINKED ISSUE #4215: Migrate API usage from *.metafi and *.metaswap subdomains to *.api.cx.metamask.io
### What is this about?

API Platform team intends to deprecate *.metafi and *.metaswap subdomains in favour of *.api.cx.metamask.io
The new *.api.cx.metamask.io is already in place.
The old *.metafi and *.metaswap are still used across the controllers.
Controllers do not break currently because redirects are in place for the older sub-domains.
The redirects will be removed once all clients (Extension, Mobile, Core) are migrated.

### Scenario

- NA

### Design

- NA

### Technical Details

- TBD

### Threat Modeling Framework

- NA

### Acceptance Criteria

- All occurrences of *.metafi and *.metaswap subdomains replaced by *.api.cx.metamask.io subdomain in the core codebase.

### Stakeholder review needed before the work gets merged

- [X] Engineering (needed in most cases)
- [ ] Design
- [ ] Product
- [ ] QA (automation tests are required to pass before merging PRs but not all changes are covered by automation tests - please review if QA is needed beyond automation tests)
- [ ] Security
- [ ] Legal
- [ ] Marketing
- [ ] Management (please specify)
- [ ] Other (please specify)

### References

- [Slack thread](https://consensys.slack.com/archives/C03MLR70YSK/p1712691934915449?thread_ts=1712687415.554859&cid=C03MLR70YSK)
- [Same issue for Extension](https://github.com/MetaMask/MetaMask-planning/issues/2351)
-[ Mobile PR ](https://github.com/MetaMask/metamask-mobile/pull/9388)

### PR DESCRIPTION
**Pull request by cryptodev-2s**
## Explanation

API Platform team intends to deprecate *.metafi and *.metaswap subdomains in favour of *.api.cx.metamask.io
The new *.api.cx.metamask.io is already in place.
The old *.metafi and *.metaswap are still used across the controllers.
Controllers do not break currently because redirects are in place for the older sub-domains.
The redirects will be removed once all clients (Extension, Mobile, Core) are migrated.

## References

* Fixes #4215

### `@metamask/assets-controller`

- **CHANGED:** Update price API endpoint from `price-api.metafi.codefi.network` to `price.api.cx.metamask.io`
- **CHANGED:** Update token icons API endpoint from `*.metafi.codefi.network` to `*.cx.metamask.io`
- **CHANGED:** Update toke API endpoint from `token-api.metaswap.codefi.network` to `token.api.cx.metamask.io`

### `@metamask/phishing-controller`

- **CHANGED:** Update phishing detection API endpoint from `*.metafi.codefi.network` to `*.api.cx.metamask.io`

### `@metamask/controller-utils`

- **CHANGED:** Update OpenSea proxy endpoint from `*.metafi.codefi.network` to `*.api.cx.metamask.io`

### `@metamask/name-controller`

- **CHANGED:** Update token API endpoint from `token-api.metaswap.codefi.network` to `token.api.cx.metamask.io`

## Checklist

- [ ] I've updated the test suite for new or updated code as appropriate
- [ ] I've updated documentation (JSDoc, Markdown, etc.) for new or updated code as appropriate
- [ ] I've highlighted breaking changes using the "BREAKING" category above as appropriate

### CODE REVIEW

File: packages/assets-controllers/src/TokensController.test.ts [hunk 1]
```diff
@@ -496,7 +490,7 @@ describe('TokensController', () => {
         address: '0x02',
         decimals: 2,
         image:
-          'https://static.metafi.codefi.network/api/v1/tokenIcons/11155111/0x02.png',
+          'https://static.cx.metamask.io/api/v1/tokenIcons/11155111/0x02.png',
```
Comment (mikesposito) [inline packages/assets-controllers/src/TokensController.test.ts:493]: I'm unable to access these static files, but I'm also unable to access them through old links. Is this expected? will these links be activated later?

**Review by mikesposito — APPROVED** [review]
Looks good! Just a small question

**Comment by mikesposito** [comment]
These changes should be transparent for external consumers, since the new endpoints should be functionally equivalent to the new ones - so I don't think we should mark them as **breaking**

**Comment by cryptodev-2s** [comment]
> These changes should be transparent for external consumers, since the old endpoints should be functionally equivalent to the new ones - so I don't think we should mark them as **breaking**.
> 
> Or do you expect something to break?

Yes I agree in somehow this should not be breaking changes but in similar change this was marked as `BREAKING` [PR](https://github.com/MetaMask/core/pull/1060)

**Comment by mcmire** [comment]
@cryptodev-2s Hmm, I am not sure why they marked the changes as breaking in that PR. That seems like a mistake to me. I had the same understanding as @mikesposito, that the new URL is simply an alias for the old one. I also agree that we should not mark these as breaking.

**Review by mcmire — APPROVED** [review]
I've cross-referenced these URL changes via Slack and they look correct to me.

**Pull request merged**